rust/hyperlane-base/src/types/multisig.rs

```diff
@@ -186,32 +220,40 @@ impl MultisigCheckpointSyncer {
                     let root = signed_checkpoint.value.root;
                     let signed_checkpoints = signed_checkpoints_per_root.entry(root).or_default();
                     signed_checkpoints.push(signed_checkpoint);
+                    cumulative_weight += vw.weight;
 
                     // Count the number of signatures for this signed checkpoint
                     let signature_count = signed_checkpoints.len();
                     debug!(
-                        validator = format!("{:#x}", validator),
+                        validator = format!("{:#x}", vw.validator),
                         index = index,
                         root = format!("{:#x}", root),
                         signature_count = signature_count,
                         "Found signed checkpoint"
                     );
 
-                    // If we've hit a quorum, create a MultisigSignedCheckpoint
-                    if signature_count >= threshold {
+                    // If we've hit a quorum in weight, create a MultisigSignedCheckpoint
+                    if cumulative_weight >= threshold_weight {
+                        // to conform to the onchain ordering of the set by address
+                        signed_checkpoints.sort_by_key(|sc| {
+                            weighted_validators
+                                .iter()
+                                .position(|vw| vw.validator == H256::from(sc.recover().unwrap()))
```

Comment:
We should avoid all unwraps at this part of the code. Just return an Err with `?` here and the line below. Even if logically we know that unwrap should never get hit because of prior logic, imo this is an unnecessary code smell and if the prior logic gets changed in the future, it won't always be obvious if it has implications further down and could result in a panic

Reply:
Ok, does unwrap_or((true, 0)) work here then?